clash-lib/src/Clash/Rewrite/Util.hs

```diff
@@ -957,12 +959,12 @@ specArgBndrsAndVars
   :: Either Term Type
   -> ([Either Id TyVar], [Either Term Type])
 specArgBndrsAndVars specArg =
-  let unitFV :: Var a -> Const (UniqSet TyVar,UniqSet Id) (Var a)
-      unitFV v@(Id {}) = Const (emptyUniqSet,unitUniqSet (coerce v))
-      unitFV v@(TyVar {}) = Const (unitUniqSet (coerce v),emptyUniqSet)
+  let unitFV :: Var a -> Const (OSet.OLSet TyVar, OSet.OLSet Id) (Var a)
+      unitFV v@(Id {}) = Const (mempty, coerce (OSet.singleton (coerce v)))
+      unitFV v@(TyVar {}) = Const (coerce (OSet.singleton (coerce v)), mempty)
 
       (specFTVs,specFVs) = case specArg of
-        Left tm  -> (eltsUniqSet *** eltsUniqSet) . getConst $
+        Left tm  -> (OSet.toListL *** OSet.toListL) . getConst $
```

Comment:
It was non-obvious why we needed an insertion-order preserving set. Please add a `-- Note [Collect free-variables in an insertion-ordered set]` that explains why.

Reply:
done